core/network/src/main/java/com/gkp/core/network/interceptor/TaskyApiInterceptor.kt

```diff
@@ -0,0 +1,26 @@
+package com.gkp.core.network.interceptor
+
+import com.gkp.core.network.AccessTokenRefreshHandler
+import com.gkp.network.BuildConfig
+import okhttp3.Interceptor
+import okhttp3.Response
+
+class TaskyApiInterceptor(
+    private val accessTokenRefreshHandler: AccessTokenRefreshHandler
+) : Interceptor {
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+            .newBuilder()
+            .addHeader(
+                "x-api-key",
+                BuildConfig.API_KEY
+            ).build()
+
+        val response = chain.proceed(request)
+        if (accessTokenRefreshHandler.shouldRefreshAccessToken(response)) {
+            accessTokenRefreshHandler.refreshAccessToken()
```

Comment:
This is the place where I'd use runBlocking instead, since here it's clear to everyone why that has to be used, while in the refresh token class it isn't